spec/mem/basic.md

```diff
@@ -118,7 +118,7 @@ impl<Provenance, Extra> Allocation<Provenance, Extra> {
         }
 
         let offset_in_alloc = addr - self.addr;
-        if offset_in_alloc < 0 || offset_in_alloc + pointee_size.bytes() > self.size().bytes() {
+        if !pointee_size.is_zero() && (offset_in_alloc < 0 || offset_in_alloc + pointee_size.bytes() > self.size().bytes()) {
```

Comment:
Why this change? This function is never even called for zero-sized pointees, I think?

Reply:
Ah, I see, this is also used directly by TB. I think that should be done differently... I have pushed 6e77e25 which avoids calling `offset_in_alloc` when the size is zero, then no change should be needed here.

Reply:
Sorry, the push failed... it's now pushed as commit 5e38d1c.